modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replication/request/ReadWriteMultiRowReplicaRequest.java

```diff
@@ -31,4 +33,12 @@ public interface ReadWriteMultiRowReplicaRequest extends MultipleRowReplicaReque
      * @return {@code True} to disable the delayed ack optimization.
      */
     boolean skipDelayedAck();
+
+    /**
+     * Optional operation types (one for every tuple in {@link #binaryTuples()}.
+     * See {@link RowOpType}.
+     *
+     * @return A byte for every tuple in {@link #binaryTuples()} indicating an operation type for that row.
+     */
+    byte @Nullable [] operationTypes();
```

Comment:
Maybe use a bitset here ? A corresponding bit is set to 1 for removal, 0 otherwise.

Reply:
Good point, done.